### PR TITLE
Enforce good practices and consistent style on tests

### PR DESCRIPTION
**Pull request by teceler**
* Uses Time.use_zone block rather than setting Time.zone
* Replaces instance variables in tests
* Uses numbers for http statuses
* Avoids expect(...).and_return(...) where it's not necessary

### CODE REVIEW

File: spec/services/post_scraper_spec.rb [hunk 1]
```diff
@@ -305,7 +305,7 @@
 
     html = "<!DOCTYPE html>\n<html></html>\n"
     stub_with_body = double
-    expect(stub_with_body).to receive(:body).and_return(html)
+    allow(stub_with_body).to receive(:body).and_return(html)
```
Comment (Throne3d) [inline spec/services/post_scraper_spec.rb:308]: What's the difference between allow and expect here? I thought expect would require a call, but allow would just stub it out (without any expectations).

**Review by Throne3d — APPROVED** [review]
Beyond the comment, LGTM!

File: spec/services/post_scraper_spec.rb (hunk 1 above)
Comment (teceler) [reply spec/services/post_scraper_spec.rb:308]: Basically that, allow stubs and expect does an expectation. (Expect also stubs In An Order which came up in one of the tests xD) Rubocop thinks you shouldn't mix expecting something and stubbing it and honestly I think it's probably right. ...possibly some of these should *also* be expected :thinking:

Comment (Throne3d) [reply spec/services/post_scraper_spec.rb:308]: How do we expect without a stub in context?

Comment (teceler) [reply spec/services/post_scraper_spec.rb:308]: I think expect to receive doesn't stub but I have to check. that or you do the stubbing desperately which is, at least, clearer as to what the heck you're doing?

**Review by Throne3d — CHANGES_REQUESTED** [review]
Spoke about this in the discord - would be good to turn the `expect(blah).to receive(thing).and_do_thing()`s into `allow(...).to receive(thing).and_to_thing()` followed by an expect on it too

File: spec/jobs/application_job_spec.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def perform(*_args)
     expect(ExceptionNotifier).to receive(:notify_exception).with(exc, data: { job: StubJob.name, args: [2, :test] })
 
     job = StubJob.new(2, :test)
-    expect(job).to receive(:perform).and_raise(exc)
+    allow(job).to receive(:perform).and_raise(exc)
```
Comment (Throne3d) [inline spec/jobs/application_job_spec.rb:31]:
```suggestion
    allow(job).to receive(:perform).and_raise(exc)
    expect(job).to receive(:perform)
```

File: spec/script/post_split_spec.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@
     let(:reply) { create(:reply) }
 
     it "requires valid subject" do
-      expect(STDIN).to receive(:gets).and_return('')
+      allow(STDIN).to receive(:gets).and_return('')
```
Comment (Throne3d) [inline spec/script/post_split_spec.rb:15]:
```suggestion
      allow(STDIN).to receive(:gets).and_return('')
      expect(STDIN).to receive(:gets)
```

File: spec/services/post_scraper_spec.rb (hunk 1 above)
Comment (Throne3d) [inline spec/services/post_scraper_spec.rb:308]:
```suggestion
    allow(stub_with_body).to receive(:body).and_return(html)
    expect(stub_with_body).to receive(:body)
```

**Review by Throne3d — COMMENTED** [review]
I think this is all!

**Review by Throne3d — APPROVED** [review]
Looks good!

**Pull request merged**